lib/host/wasi_nn/wasinnfunc.cpp

```diff
@@ -3,49 +3,474 @@
+#ifdef WASMEDGE_WASINN_BUILD_OPENVINO
+    if (BuilderLen != 2) {
+      spdlog::error("[WASI-NN] Wrong GraphBuilder Length {:d}, expecting 2",
+                    BuilderLen);
+      return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+    }
+    uint32_t *GraphBuilders =
+        MemInst->getPointer<uint32_t *>(BuilderPtr, BuilderLen * 2);
+    uint32_t *GraphId = MemInst->getPointer<uint32_t *>(GraphPtr, 1);
+
+    IEStatusCode Status;
+    if (Ctx.OpenVINOCore == nullptr) {
+      Status = ie_core_create("", &Ctx.OpenVINOCore);
+      if (Status != IEStatusCode::OK || Ctx.OpenVINOCore == nullptr) {
+        spdlog::error("[WASI-NN] Error happened when init OpenVINO core.");
+        return static_cast<uint32_t>(WASINN::ErrNo::MissingMemory);
+      }
+      spdlog::debug("[WASI-NN] Initialize OpenVINO Core");
+    }
+
+    std::string DeviceName = mapTargetToString(Target);
+    if (DeviceName.length() == 0) {
+      spdlog::error("[WASI-NN] Device target {:d} not support!", Target);
+      return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+    } else {
+      spdlog::debug("[WASI-NN] Using device: {:s}", DeviceName);
+    }
+
+    uint32_t XMLStringLen = GraphBuilders[1];
+    uint32_t WeightBinsLen = GraphBuilders[3];
+    uint8_t *XMLPtr =
+        MemInst->getPointer<uint8_t *>(GraphBuilders[0], XMLStringLen);
+    uint8_t *BinPtr =
+        MemInst->getPointer<uint8_t *>(GraphBuilders[2], WeightBinsLen);
+    tensor_desc_t WeightsDesc{
+        layout_e::ANY, {1, {WeightBinsLen}}, precision_e::U8};
+    ie_blob_t *WeightsBlob = nullptr;
+    ie_blob_buffer_t BlobCBuffer;
+
+    Status = ie_blob_make_memory(&WeightsDesc, &WeightsBlob);
+    if (Status != IEStatusCode::OK || WeightsBlob == nullptr) {
+      spdlog::error(
+          "[WASI-NN] Unable to create model's weight blob, error code: {}",
+          Status);
+      return static_cast<uint32_t>(WASINN::ErrNo::Busy);
+    }
+    Status = ie_blob_get_cbuffer(WeightsBlob, &BlobCBuffer);
+    if (Status != IEStatusCode::OK) {
+      spdlog::error(
+          "[WASI-NN] Unable to find weight blob's buffer, error code: {}",
+          Status);
+      return static_cast<uint32_t>(WASINN::ErrNo::MissingMemory);
+    }
+    uint8_t *BlobData = static_cast<uint8_t *>(BlobCBuffer.buffer);
+    for (size_t I = 0; I < WeightBinsLen; I++) {
+      BlobData[I] = BinPtr[I];
+    }
+
+    ie_network_t *Network = nullptr;
+    ie_executable_network_t *ExeNetwork = nullptr;
+    Status = ie_core_read_network_from_memory(
+        Ctx.OpenVINOCore, XMLPtr, XMLStringLen, WeightsBlob, &Network);
+    if (Status != IEStatusCode::OK || Network == nullptr) {
+      spdlog::error("[WASI-NN] Unable to create Network");
+      return static_cast<uint32_t>(WASINN::ErrNo::Busy);
+    }
+
+    size_t NetworkInputSize;
+    Status = ie_network_get_inputs_number(Network, &NetworkInputSize);
+    // FIXME this is a temporary workaround. We need a more eligant way to
```

Comment:
`FIXME:`
Don't forget the `:`. It's the format to tag the comments.

Reply:
Got it